healthyregions/SDOHPlace
# #180: create new search page

This PR is for #180. I renamed the old search index page to `index_back.tsx` and created a new index page that combines the new result page components to match the updated UI design. This PR mainly targets UI adjustments, as these changes are substantial enough to warrant their own PR.

I also fixed the input field error in this PR, which was observed in last week's demo. Since the search box has been extracted as a distinct component, some input controls needed adjustment to coordinate with this change. This PR includes that fix. Further adjustments to the input fields may be made when working on the new filter functions.

## How to Test

1. Navigate to the search page. You should see that the layout matches the design.

-----

2. The result list panel should show "All data sources" without a "Related" field since the user hasn't done any search. Note that the process bar is controlled by the browser, so it appears when the user hovers over the result list. The user should also notice that the result list doesn't end until the bottom of the page:

<img width="525" alt="Screenshot 2024-07-29 at 2 39 19 PM" src="https://github.com/user-attachments/assets/0cf68c8d-74ff-4763-bfce-33904bb82d14">
<img width="542" alt="Screenshot 2024-07-29 at 2 40 33 PM" src="https://github.com/user-attachments/assets/bc19347a-b800-49a1-930f-309867692e02">

The map side should have headers and the bottom description part added. The two buttons on the bottom of the map side should link to the correct places:

<img width="881" alt="Screenshot 2024-07-29 at 2 44 38 PM" src="https://github.com/user-attachments/assets/ac04ca1e-2bd1-455d-9871-995c56e74653">

-----

3. Search for anything in the input field. You should see the result panel change to "Results" as the header, and a "Related" field should be added (currently, the result list is used since we haven't decided what should be categorized as "related"):

<img width="1588" alt="Screenshot 2024-07-29 at 2 47 30 PM" src="https://github.com/user-attachments/assets/6f6e740b-02a6-4600-8e81-b62be105d072">

Note that if your search results contain more than three items (e.g., "Wisconsin" or just an "a"), the result list will be scrollable, and the height of the map will match the height of the result list:

<img width="1491" alt="Screenshot 2024-07-29 at 3 53 16 PM" src="https://github.com/user-attachments/assets/a0e344a4-e3f6-4a77-8ac7-3ce9a20c7c74">

-----

4. Gradually press the 'delete' key to remove all characters in the input field, or select all characters in the input field and delete them at once. You should notice that once all characters are removed, the page resets to step 1 (but doesn't affect the spatial resolution status):

<img width="778" alt="Screenshot 2024-07-29 at 2 48 37 PM" src="https://github.com/user-attachments/assets/a6587274-9dee-4f38-97bd-9807bb91d2cf">

-----

5. Click any "View" for a result list item. You should notice the corresponding item gets highlighted, and you can review the details on the right panel:

<img width="646" alt="Screenshot 2024-07-29 at 2 50 28 PM" src="https://github.com/user-attachments/assets/66ecad3f-cf2e-40d5-9c38-387a77ae7900">
